### PR TITLE
Vcs config option

### PR DESCRIPTION
**Pull request by wesleywiser**

### CODE REVIEW

**Comment by wesleywiser** [comment]
Fixes #1116

File: src/cargo/ops/cargo_new.rs [hunk 1]
```diff
@@ -158,12 +185,20 @@ fn global_config() -> CargoResult<CargoNewConfig> {
...
+        Some(vcs) => {
+            let vcs_str = try!(vcs.string().chain_error(|| {
+                internal("invalid configuration for key `cargo-new.vcs`")
+            })).0;
+            let version_control = match vcs_str.as_slice() {
+              "git" => VersionControl::Git,
+              "hg"  => VersionControl::Hg,
+              "none"=> VersionControl::NoVcs,
+              _  => return Err(internal("invalid configuration for key `cargo-new.vcs`")),
+            };
```
Comment (alexcrichton) [inline src/cargo/ops/cargo_new.rs]: Typically we use 4-space tabs for indentation, could you swap this and some parts up above to 4-space tabs?

File: src/bin/new.rs [hunk 2]
```diff
@@ -24,10 +22,9 @@ Usage:
 
 Options:
     -h, --help          Print this message
-    --no-git            Don't initialize a new git repository
-    --git               Initialize a new git repository, overriding a
-                        global `git = false` configuration
-    --hg                Initialize a new hg repository
+    --vcs <vcs>         Initialize a new repository for the given version
+                        control system (git or hg) or do not initialize any version 
+                        control at all overriding a global configuration.
```
Comment (alexcrichton) [inline src/bin/new.rs]: Could this also mention `none`?

File: src/cargo/ops/cargo_new.rs [hunk 3]
```diff
@@ -2,24 +2,41 @@ use std::os;
...
 }
 
+impl<E, D: Decoder<E>> Decodable<D, E> for VersionControl {
+    fn decode(d: &mut D) -> Result<VersionControl, E> {
+        Ok(match try!(d.read_str()).as_slice() {
+            "git" => VersionControl::Git,
+            "hg" => VersionControl::Hg,
+            "none" => VersionControl::NoVcs,
+            n => {
+                let err = format!("Could not decode '{}' as version control.", n);
```
Comment (alexcrichton) [inline src/cargo/ops/cargo_new.rs]: Our errors typically start with a lowercase letter and don't end with a period.

**Comment by alexcrichton** [comment]
This looks great, thanks @wesleywiser! `--vcs none` is a clever idea!

I think this also closes #1099 and at least part of #517 for now.

File: src/cargo/ops/cargo_new.rs (hunk 1 above)
Comment (wesleywiser) [inline src/cargo/ops/cargo_new.rs]: Absolutely

File: src/bin/new.rs (hunk 2 above)
Comment (wesleywiser) [inline src/bin/new.rs]: Good idea.

**Comment by wesleywiser** [comment]
Thanks! Is there any documentation I should update as part of these changes?

File: src/cargo/ops/cargo_new.rs (hunk 3 above)
Comment (wesleywiser) [inline src/cargo/ops/cargo_new.rs]: Will fix.

**Comment by alexcrichton** [comment]
Ah yes I think there is, if you grep for `--git` inside of `src/doc` you may find a hit or two.

**Comment by wesleywiser** [comment]
Done

**Pull request merged**